OfflineAudioContext/explainer.md

```diff
@@ -0,0 +1,203 @@
+
+The [spec](https://webaudio.github.io/web-audio-api/#dom-offlineaudiocontext-startrendering) explicitly states at step 5: "Create a new AudioBuffer ... with ... length and sampleRate ... Assign this buffer to an internal slot" which means the API design currently mandates the full buffer allocation.
+
+The participants on the [GitHub discussion](https://github.com/WebAudio/web-audio-api/issues/2445) agree that incremental delivery of data is necessary. Either streaming chunks of rendered audio or dispatching data in bits rather than everything at once so that memory usage is bounded and the data can be processed/consumed as it is produced.
+
+## User-Facing Problem
+
+The user in this context is the web developer using the WebAudio API to perform media processing workflows. Ideally developers could use the feature-rich WebAudio API for realtime and faster-than-realtime processing, without taking a dependency on a 3rd party library. However, in reality, the current WebAudio OfflineAudioContext API is not suitable for faster-than-raltime processing so the developer needs to create a WASM audio processing library or use an existing 3rd party dependency to achieve this goal.
+
+### Goals
+
+- Allow streaming data out of a WebAudio `OfflineAudioContext.startRendering()` for rendering large WebAudio graphs faster-than-realtime
+
+### Non-goals
+
+- Change the existing `startRendering()` behavior, this API change is additive
+
+## Proposed Approach
+
+The preferred approach is to allow `startRendering()` to be configured to stream output via a `StartRenderingOptions` object which will have a `mode` property. This `mode` can be set to `"stream"` which will stream data rather than allocating an `AudioBuffer` up front.
+
+The object allows for future flexibility of the new API surface.
+
+```typescript
+interface StartRenderingOptions {
+    mode: "audiobuffer" | "stream"
+}
+
+interface OfflineAudioContext {
+  startRendering(options?: startRenderingOptions}
+  ): Promise<AudioBuffer | ReadableStream | void>;
+}
+```
+
+If the `startRendering` function is passed `{ mode = "stream" }` then it will render the audio graph in quantums (e.g., 128 frames at a time), and enqueue chunks onto a return `ReadableStream`, rather than rendering the whole graph into an `AudioBuffer` up front as it does currently. `startRendering` will return a [ReadableStream](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/getReader) promise. A `reader` can be retrieved off the `ReadableStream` for reading chunks. `reader.read()` will resolve stream values to an `AudioBuffer` until it is done. When no more data is available it will set `done = true`.
```

Comment:
Maybe we can remove the last sentence? "When no more data is available it will set `done = true`". It is not clear what it refers to.